app/model/repository/Groups.php

```diff
@@ -26,13 +26,16 @@ public function __construct(EntityManagerInterface $em)
     /**
      * Find all groups belonging to specified instance.
      * @param Instance $instance
-     * @return array
+     * @return Group[]
      */
     public function findAllByInstance(Instance $instance)
```

Comment:
Wouldn't it be better to also add `array` type hint for the return value? (same goes for some functions below in this file)

Reply:
Ok, will add